src/api/entities/SecurityToken/TokenHolders.ts

```diff
@@ -27,16 +42,56 @@ export class TokenHolders extends Namespace<SecurityToken> {
 
     const { entries, lastKey: next } = await requestPaginated(query.asset.balanceOf, {
       arg: rawTicker,
-      paginationOpts,
+      paginationOpts: opts?.pagination || undefined,
     });
 
-    const data: IdentityBalance[] = entries.map(([storageKey, balance]) => ({
-      identity: new Identity(
-        { did: identityIdToString(storageKey.args[1] as IdentityId) },
-        context
-      ),
-      balance: balanceToBigNumber(balance),
-    }));
+    let data: IdentityBalance[] = [];
+
+    const securityToken = new SecurityToken({ ticker }, context);
+
+    const areFrozen = opts?.mintStatus ? await securityToken.transfers.areFrozen() : undefined;
+
+    if (opts?.mintStatus && !areFrozen) {
+      const entriesChunks = chunk(entries, 10);
+
+      await Promise.all(
```

Comment:
This one (and the one in issueTokens as well) should use Bluebird's `.each` here instead of Promise.all (my bad). The idea is to process each chunk serially, otherwise there is no point in separating them into chunks in the first place. The one in the inside is fine, as we want all elements in each chunk to be processed simultaneously